src/IW/App/Error.hs

```diff
@@ -109,6 +112,27 @@ data IError
     | DbError Text
     deriving (Show, Eq)
 
+{- | Errors from the @github@ library search functions that can be thrown. 
+-}
+data GError 
+    {- | A HTTP error occurred. The actual caught error is included. -}
+    = HTTPError Text	
+    {- | An error in the parser itself. -}
+    | ParseError Text	
+    {- | The JSON is malformed or unexpected. -}
+    | JsonError Text	
+    {- | Incorrect input was provided. -}
+    | UserError Text
+    deriving (Show, Eq)
+
+-- | Map the @github@ library's @Error@ type into AppErrorType.
+githubErrToAppErr :: GitHub.Error -> AppErrorType
+githubErrToAppErr err = case err of
+    GitHub.HTTPError httpException -> GitHubError . HTTPError $ show httpException 
+    GitHub.ParseError text         -> GitHubError . ParseError $ text 
```

Comment:
I personally don't like `f . g . h $ x` file. I find that it requires to type redundant characters. Something like the following looks simpler to me:

```haskell
GitHubError $ ParseError text
```

Never had problems with this, and I don't know why some people in community force this `. . . $` style :slightly_smiling_face:

Reply:
Got it👍🏽